udfs/remote_udfs/nlp/README.md

```diff
@@ -0,0 +1,103 @@
+# Calling Google's Natural Language API Sentiment Analysis Example 
+
+Google's Natural Language API Sentiment Analysis inspects the given text and identifies the prevailing emotional opinion within the text, especially to determine a writer's attitude as positive, negative, or neutral.  
+
+To run this example, you will need the following APIs enabled:
+* [BigQuery API](https://cloud.google.com/bigquery/docs)
+* [Cloud Functions API](https://cloud.google.com/functions/docs)
+* [BigQuery Connections API](https://cloud.google.com/bigquery/docs/working-with-connections#enable_the_connection_service)
+* [Natural Language API](https://cloud.google.com/natural-language/docs/setup#api)
+
+## Running the example
+
+### Creating your BigQuery connection 
+
+To create your BigQuery connection, you need to specify the following:
+* project_id - the project id that you wish to create the connection in
+* location - the location for the external connection in BigQuery
+* connection_name - the name of the connection 
+* display_name - (optional) the name (friendly name) you want to show 
+
+```
+bq mk --connection --display_name=\'$DISPLAY_NAME\' --connection_type=CLOUD_RESOURCE \
+      --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME
+```
+
+### Getting the service account associated with the connection
+
+You need to grant the connection service account to be able to invoke functions.  
+To do this, first obtain the service account. 
+* location - the location for the external connection in BigQuery
+* project - the project where your external connection was created
+* connection_name - the name (or friendly/display name) of the external connection
+
+```
+bq ls --connection --location=$LOCATION
+bq show --connection --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME
+```
+
+Within properties you'll find *serviceAccountId* which will have the service ID you'll need in a subsequent step.
+
+### Deploying your Cloud Function
+
+[More information about deploying your cloud function can be found here.](https://cloud.google.com/functions/docs/deploy)
+
+Snippet provided below for brevity.
+* cf_name - the name of the cloud function, this should match the entry function in your code (in the example it's *remote_vertex_ai*)
+* project - the project the cloud function is deployed to 
+* runtime - this was defaulted to python39 but can be changed as required 
+```
+gcloud functions deploy $CF_NAME \
```

Comment:
This is all under the "Running the example" section.
Make it easy for a user to run the example as-is, so set the values for things like CF_NAME

Reply:
Also, CF_NAME and the function entry do not have to be the same, and if anything it's confusing to recommend it because users may deploy several functions from the same source with different entries. 

Instead add [--entry-point option](https://cloud.google.com/sdk/gcloud/reference/functions/deploy#--entry-point)

Reply:
The only thing I can't do is the service account since it's created after the fact - (I can script it out) let me know what your thoughts are.

Reply:
`gcloud functions describe remote_vertex_ai --format="value(httpsTrigger.url)"`

Reply:
Yeah, I can script it out, sorry that's what I meant, then I'll do that

Reply:
for the bq tool, you'll have to use jq to parse out the serviceAccount as follows:

`bq show --connection --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME | jq '.cloudResource.serviceAccountId'`

Reply:
updated